compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUSelectLoweringStrategy.cpp

```diff
@@ -45,11 +45,14 @@ verifyLoweringConfiguration(FunctionOpInterface funcOp,
                             IREE::Codegen::TranslationInfoAttr translationInfo,
                             ArrayRef<int64_t> workgroupSize, F verificationFn) {
   auto walkResult = funcOp.walk([&](Operation *op) -> WalkResult {
-    auto loweringConfig =
+    auto codegenLoweringConfig =
         getLoweringConfig<IREE::Codegen::LoweringConfigAttr>(op);
-    if (!loweringConfig)
+    auto gpuLoweringConfig =
+        getLoweringConfig<IREE::GPU::LoweringConfigAttr>(op);
```

Comment:
Rather than casting and passing two lowering config attributes, it would be cleaner to add an additional template parameter to this function (`template <typename F, typename ConfigTy>`) and only cast to the config type that works for `F`. Then rather than combining the verifiers for `LLVMGPUMatmulTensorCoreMmaSync` and `LLVMGPUVectorDistribute`, we have two separate `verifyGPUMatmulPipeline` functions, one for the GPU lowering config and one for the Codegen lowering config.

Reply:
@qedawkins, Are you ok for the last updates? A Little different  than yours.